src/packet/bwe/acked_bitrate_estimator.rs

```diff
@@ -0,0 +1,161 @@
+use std::time::{Duration, Instant};
+
+use crate::rtp::DataSize;
+use crate::Bitrate;
+
+pub struct AckedBitrateEstimator {
+    /// The initial window to use for the first estimate.
+    initial_window: Duration,
+    /// The window to use for subsequent estimates after achieving the initial estimate.
+    window: Duration,
+    /// The estimate of the acked bitrate.
+    estimate: Option<Bitrate>,
+    /// The sum in the current window.
+    sum: DataSize,
+    /// The size of the current window.
+    current_window: Duration,
+    /// The last time the window was updated.
+    last_update: Option<Instant>,
+}
+
+impl AckedBitrateEstimator {
+    pub(super) fn new(initial_window: Duration, window: Duration) -> Self {
+        Self {
+            initial_window,
+            window,
+            estimate: None,
+            sum: DataSize::ZERO,
+            current_window: Duration::ZERO,
+            last_update: None,
+        }
+    }
+
+    pub(super) fn update(&mut self, receive_time: Instant, packet_size: DataSize) {
+        let window = if self.estimate.is_none() {
+            // Use the initial, larger, window at first
+            self.initial_window
+        } else {
+            self.window
+        };
+        let Some(update) = self.update_window(receive_time, packet_size, window) else {
+            // No update
+            return;
+        };
+
+        self.estimate = Some(update);
+    }
+
+    pub(super) fn current_estimate(&self) -> Option<Bitrate> {
+        self.estimate
+    }
+
+    fn update_window(
+        &mut self,
+        receive_time: Instant,
+        packet_size: DataSize,
+        window: Duration,
+    ) -> Option<Bitrate> {
+        if self
+            .last_update
+            .map(|ls| receive_time < ls)
+            .unwrap_or(false)
```

Comment:
I think this could be clearer with

```rust
let time_moved_back = Some(receive_time) < self.last_update;
if time_moved_back {
…
}
```

Reply:
This is also a case where `will_not_happen()` as initial value will make easier logic than juggling that `Option`.

Reply:
Are you sure, it would need to be `already_happened` but then I do math on this so that would need special handling to not calculate a window that's several hours.

I did what you suggested above